File: nhs_backend/java/src/main/java/com/bhtask/nhs/backend/controller/ConditionsController.java
package com.bhtask.nhs.backend.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.bhtask.nhs.backend.dao.Condition;
import com.bhtask.nhs.backend.service.ConditionsService;

@RestController
@RequestMapping("/conditions")
public class ConditionsController {

    @Autowired
    private ConditionsService conditionsService;

    @GetMapping("/alive")
    public String alive() {
	return "Yes!";
    }

    @GetMapping("/all")
    public List<String> all() {
	return conditionsService.getAllLinks();
    }

    @GetMapping("/count")
    public Long count() {
	return conditionsService.getCount();
    }

    @GetMapping("/search")
    public List<Condition> search(@RequestParam("query") final String query,
	    @RequestParam(name = "includeContent", required = false, defaultValue = "true") boolean hideContent) {

	List<Condition> results = conditionsService.search(query);
	return results;
    }
}

File: nhs_backend/scala/src/main/scala/com/bhtask/nhs/backend/controller/ConditionsController.scala
package com.bhtask.nhs.backend.controller

import java.util

import com.bhtask.nhs.backend.dao.Condition
import com.bhtask.nhs.backend.service.ConditionsService
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation._

import scala.collection.JavaConverters._

@RestController
@RequestMapping(path = Array("/conditions"))
class ConditionsController(@Autowired val conditionsService: ConditionsService) {

  private final val logger: Logger = LoggerFactory.getLogger(classOf[ConditionsController])

  @GetMapping(path = Array("/alive"))
  def alive(): String = "Yes"

  @GetMapping(path = Array("/all"))
  def all(): util.Collection[String] = {
    logger debug "Received request for all conditions links."
    conditionsService.getAllLinks.asJavaCollection
  }

  @GetMapping(path = Array("/count"))
  def count(): Long = {
    logger debug "Received request for conditions count."
    conditionsService getCount
  }

  @GetMapping(path = Array("/search"))
  def search(@RequestParam query: String): util.Collection[Condition] = {
    logger debug "Received search query for conditions."
    conditionsService.search(query).asJavaCollection
  }

  @PostMapping(path = Array("/loadIndex"))
  def loadIndex(): String = {
    logger debug "Received request for loading data into search index."
    if (conditionsService loadIndex) {
      "Succeeded!"
    } else {
      "Failed to (re)load data into index."
    }
  }
}
